### ISSUE TITLE
Add README guidance for `.jekyll-metadata`

### ISSUE THREAD

**Issue by fj** [jekyll/jekyll#3595]
As of 3.0.0 Jekyll now generates a `.jekyll-metadata` file, but it's not clear if this is something that should be in source control or not. It's not in the [official site](https://github.com/jekyll/jekyll/tree/master/site), for example. Can a section be added to the README explaining this?

**Comment by parkr** [comment]
@fj Is the readme the best place? Maybe an `Upgrading to 3.0` doc?

**Comment by pathawks** [comment]
It will be relevant even to users who start using Jekyll after v3, so I think an `Upgrading to 3.0` is *not* the best place.

**Comment by parkr** [comment]
Where do we indicate that you shouldn't track `_site`?

**Comment by pathawks** [comment]
> Where do we indicate that you shouldn't track `_site`?

I'm not sure I understand.

[Upgrading](http://jekyllrb.com/docs/upgrading/) has information about changes to Jekyll commands from v1 to v2, but all of that information is also on the [Basic Usage](http://jekyllrb.com/docs/usage/) page.

If I were new to Jekyll and wanted to learn more about this `.jekyll-metadata` file, the first place I would check is probably the [Directory structure](http://jekyllrb.com/docs/structure/) page.

**Comment by parkr** [comment]
This issue is about "hey, we should let people know they should ignore the .jekyll-metadata file in their source control system", not "hey, what is this new file? tell me about it!".

We generally say "don't put `_site` under source control." I was wondering if we said that anywhere on our site. That would be a natural place to say "oh, also ignore this other thing."

**Comment by pathawks** [comment]
*Oh* ok. :ok_hand:

Yes, that is on the [Directory structure](http://jekyllrb.com/docs/structure/) page:

> **`_site`**
> This is where the generated site will be placed (by default) once Jekyll is done transforming it. It’s probably a good idea to add this to your `.gitignore` file.

**Comment by parkr** [comment]
Ok, let's add it there then.

**Comment by pathawks** [comment]
I can work on that.

Are there currently any docs regarding incremental regeneration?

**Comment by parkr** [comment]
Not sure. @alfredxing, do you know?

**Comment by chrisfinazzo** [comment]
@pathawks,

I don't see anything about incremental regeneration when searching inside of `site/_docs`. Perhaps (if understand your suggestion correctly) an additional item on the files/directory table is the way to go.

@alfredxing,

Did this get added in one of the PR's for incremental re-gen in the prerelease builds? Haven't checked out them myself, so it may already exist and this change can just land there when it's ready.

**Comment by alfredxing** [comment]
@parkr I don't think there are any docs... I'll need to work on that. Where do you think would be the best place to put regen docs?

Looking at the `Directory structure` page, it does seem like the right place to put the info regarding `.jekyll-metadata`.

**Comment by parkr** [comment]
Let's open another issue for the regen docs :)

**Comment by jakebarnett** [comment]
^did that issue get opened/do those docs exist?

**Comment by envygeeks** [comment]
@jakebarnett I don't recall a ticket ever being created for that but I could be wrong.

**Comment by tomjohnson1492** [comment]
It's so funny -- you guys are debating about where the most likely place is to put info about .jekyll-metadata. But when I googled "what is jekyll-metadata" I landed on this page. The more you talk about where to put doc for a feature, the more make it so Google finds your discussion page about the feature, instead of the target page for it.

**Comment by envygeeks** [comment]
@tomjohnson1492 so your logic is to add to the discussion and help that bias? Even if your actually wrong about how Google works, that is your logic, to do what you are complaining about?

**Comment by tomjohnson1492** [comment]
you're right -- according to my logic, I'm just making it worse. :)

**Comment by envygeeks** [comment]
To add guidance @tomjohnson1492 whether you add `.jekyll-metadata` is really a personal choice since it's a marshaled data file with things like last modification time, if you use a build server to build your data it might actually be useful to commit it.  On our build servers we actually do save that file if the user does not provide it because it speeds up TTL for them.
